components/FetchedDogs/FetchedDogs.tsx

```diff
@@ -1,121 +1,59 @@
-import React, { useState, useEffect, useRef, useCallback } from "react";
-import { useLocalStorage } from "../../hooks/useLocalStorage";
+import React, { useEffect, useState, useRef, useCallback } from "react";
 
-import {
-  DogDetails,
-  DogsDetails,
-  likedDogsDetails,
-  useAppContext,
-} from "../../context";
+import { likedDogsDetails, useAppContext } from "../../context";
 
 import { HeartButton } from "../Buttons/HeartButton/HeartButton";
 import { LoadingSpinner } from "../LoadingSpinner/LoadingSpinner";
 
 import style from "./FetchedDogs.module.css";
-import { Breed } from "../../data/breedsData";
-
-type fetchedDog = {
-  message: string;
-  status: string;
-};
+import Link from "next/link";
 
 export const FetchedDogs = () => {
   const {
     loading,
-    setLoading,
-    fetchedDogs,
-    setFetchedDogs,
+    error,
+    dogs,
+    loadMore,
+    isMosaic,
     likedDogs,
     setLikedDogs,
-    setIsError,
-    apiCallCounter,
-    setApiCallCounter,
+    storagedBreeds,
   } = useAppContext();
 
-  const [breed, setBreed] = useState<Breed>();
-  const [random, setRandom] = useState(0);
-  const [storagedBreeds] = useLocalStorage<Breed[]>("selectedBreeds", []);
-
+  const [isScreenFilled, setIsScreenFilled] = useState(false);
   const observer = useRef<null | IntersectionObserver>(null);
 
-  const getRandomBreed = () => {
-    setRandom(Math.floor(Math.random() * storagedBreeds.length));
+  const waitForAll = () => {
```

Comment:
Is this even needed? Is see some issues here:
- the function is recreated every render
- **the function is called every render**
- **you setup a new setTimeout every render**
- **you don't clear already set timeouts**.
E.g. If the page will rerender 3 times, and I will go to another one, before timeout, those 3 setTimeouts will throw errors. It's due to the fact that the page component won't exist anymore so do the `setIsScreenFilled` method. Not to mention possible performance problems.

The only possible way would be to use this inside useEffect once and to clear already set timeout on component unmount. But as I said, probably you don't need this method at all, think about this.